SEP-0009.md

```diff
@@ -0,0 +1,102 @@
+beginning of every year, if it differs from this recommendation. Releases are
+recommended to be timed to pertinent release cycles of upstream packages. The
+lead developer can at their discretion do an out of cycle release if they feel
+there is sufficient justification.
+
+## Supporting Releases with bug fixes
+
+This SEP defines two categories of SunPy release, a Long
+Term Support (LTS) release and a short-support or non-LTS release. Both types of
+releases will be supported by bug fixes and documentation upgrades but for
+differing amounts of time. The first release
+of the year shall be a non-LTS release, which shall be supported
+for 6 months or until the next release. The second release of the year
+will be an LTS release and will be supported for 12 months or until
+the next LTS release. Support periods can be extended beyond the requirements
+here if needed.
+
+The objective of this support cycle is to balance the desire to provide a
+release cycle for people who wish to experience minimal changes to SunPy with
+minimizing the maintenance burden on the SunPy developers. This release schedule
+is consistent with the resources available at the time of writing. Should more
+(or less) resources become available this topic must be re-evaluated.
+
+## Release Numbering
+
+The following versioning system shall be used.
+
+X.Y.z
+
+Where the three components have the following meanings:
+
+* X. is the LTS version number, this will be incremented with every LTS.
+* Y. is the release counter, this will be 0 for LTS releases and increment for each intermediate release.
+* z. is the bug fix number, and is to be incremented for any bug fix releases.
+
+An example release sequence, following this scheme, would look like this:
+
+* 1.0.0 is an LTS release, released November 2019.
+* 1.0.1 is a bugfix to the LTS release, released December 2019.
+* 1.1.0 is a short support release, released May 2020.
+* 1.0.2 and 1.1.1 are bugfix releases, released June 2020, to fix a current bug and backport the fix to the LTS version.
+* 1.2.0 is an unplanned short support release, released August 2020
+* 2.0.0 is an LTS release, released November 2020.
+
+Every numbered release shall be associated with a Digital Object Identifier
+(DOI) and it is recommended that major LTS release be supported by journal
+article if appropriate.
+
+## Deprecations and Documentation
+
+The SunPy core package must always document API changes, in the CHANGELOG and in
+the user documentation. Where practical a side-by-side comparison of old and new
+functionality will be provided.
+
+In addition to this, where possible, code will emit deprecation warnings
+to inform users of planned changes to or removal of API. Warnings will **only be
+emitted** when an alternative option exists for the functionality, or the
+functionality is to be completely removed.
+
+All effort will be made to provide at least one LTS release where functionality
```

Comment:
This change came out of a conversation with @dpshelio 

@nabobalis what's your thoughts on this?

Reply:
I'd ask someone smarter.

Reply:
I don't totally get what you mean in the paragraph, basically you pledge to get at least a 2 year deprecation cycle?

Reply:
Shall we have a time commitment here instead of "one LTS release"? In case the frequency of LTS releases changes...

Reply:
The purpose of this is to ensure that if any users go from LTS to LTS release they still get a deprecation warning about pending changes. So with the planned release cadence it would be a one year deprecation cycle.

@tiagopereira What do you see as the value in a time based deprecation policy? I think that if the cadence of the LTS releases change then the deprecation time would also want to change along with it, so that as I mentioned people going from LTS to LTS will always get a warning.

Reply:
I'm not sure if a time-based policy is the way to go. I just wanted to ensure that API changes do not happen very often, in case LTS releases become very frequent. Probably fine to keep the current wording.

Reply:
The idea of this SEP isn't to overly restrict the ability for us to make API changes, but the idea of an LTS is that you get at least a year of support on that release, so you can opt into slow API changes by sticking on that LTS and then moving onto whatever the current LTS is after a year.

Reply:
>LTS releases become very frequent

By definition they cannot be that frequent, from the practical point of view sunpy is manure enough that no one wants to support multiple releases for years.

@Cadair - I missed the point that LTSes are proposed to be a yearly thing, a minimum ~1-year deprecation seems totally reasonable then.